torch/_decomp/decompositions.py

```diff
@@ -268,6 +307,49 @@ def log_sigmoid_backward(grad_output: Tensor, self: Tensor, buffer: Tensor) -> T
     # return (max_deriv - sign * (buffer / (1 + buffer))) * grad_output
 
 
+def apply_loss_reduction(loss: Tensor, reduction: int):
+    if reduction == Reduction.MEAN.value:
+        return torch.mean(loss)
+    elif reduction == Reduction.SUM.value:
+        return torch.sum(loss)
+    else:
+        return loss
+
+
+def to_real_dtype(dtype: torch.dtype):
+    if dtype == torch.complex32:
+        return torch.float16
+    elif dtype == torch.complex64:
+        return torch.float32
+    elif dtype == torch.complex128:
+        return torch.float64
+
+
+@register_decomposition(aten.l1_loss)
+def l1_loss(self: Tensor, target: Tensor, reduction: int = Reduction.MEAN.value) -> Tensor:
```

Comment:
it shouldn't be necessary to specify default arguments explicitly here; the decomp runner should fill them in automatically from schema

Reply:
Is there any reason to not do it? I could imagine in other situations (such as directly using it), the user might not have the defaults.

Reply:
I suppose the hazard is mostly violation of DRY. I guess if our defaults never change it is relatively harmless.

Reply:
If we had a (from python) way of calling an operator with the decompositions filled in, I'd be happy to remove them. 

I'd also be happy to remove them if they ever cause us problems in the future lol.